In [71]:
import math
import numpy as np

x = [(1/10), (2/10),(3/10),(4/10)]
j = [1,2,3,4]
numer = np.zeros(test)
def taylorSeries(x):
    taylor = 0 
    sign = 1
    n = 0
    while True:
        taylor = taylor + sign * (x**n) / math.factorial(n) #taylor polynomia
        error = taylor - np.cos(x) #|T(x)-cos(x)|
        if abs(error) < (10**(-12)): #absoulte valute of the error
            break
            
        
        sign *= -1 
        factorial = math.factorial(n)
        n += 2
        
        
        print("x =", x*10)
        print("Taylor =", taylor)
        print("exact cosine =", np.cos(x))
        print("n =", n)
        print("error =", error, "\n")
    
    
    
        
    
    
for j in range(len(j)+1):
    taylorSeries(j/10)


x = 1.0
Taylor = 1.0
exact cosine = 0.9950041652780258
n = 2
error = 0.0049958347219741794 

x = 1.0
Taylor = 0.995
exact cosine = 0.9950041652780258
n = 4
error = -4.165278025825003e-06 

x = 1.0
Taylor = 0.9950041666666667
exact cosine = 0.9950041652780258
n = 6
error = 1.3886408689600671e-09 

x = 2.0
Taylor = 1.0
exact cosine = 0.9800665778412416
n = 2
error = 0.019933422158758374 

x = 2.0
Taylor = 0.98
exact cosine = 0.9800665778412416
n = 4
error = -6.657784124164401e-05 

x = 2.0
Taylor = 0.9800666666666666
exact cosine = 0.9800665778412416
n = 6
error = 8.882542501531532e-08 

x = 2.0
Taylor = 0.9800665777777777
exact cosine = 0.9800665778412416
n = 8
error = -6.346390080125275e-11 

x = 3.0
Taylor = 1.0
exact cosine = 0.955336489125606
n = 2
error = 0.04466351087439402 

x = 3.0
Taylor = 0.955
exact cosine = 0.955336489125606
n = 4
error = -0.00033648912560602096 

x = 3.0
Taylor = 0.9553375
exact cosine = 0.955336489125606
n = 6
error = 1.0108743939696296e-06 

x = 3.0
Taylo

| $X$ | Taylor | Exact Cosine | $N$ | Error | Remainder Term    
| :---: | :---: | :---: | :---: | :---: | :---: 
|$1$| 0.9950041652777778  | 0.9950041652780258 | $6$ | -2.4802382370125997e-13 | 1.980821758865639e-12 
|$2$| 0.9800665778412698  | 0.9800665778412416 | $8$ | 2.8199664825478976e-14 | -2.8030946143923996e-13
|$3$| 0.9553364891256049  | 0.955336489125606 | $10$ | -1.1102230246251565e-15 | 1.311490852208501e-14
|$4$| 0.92106099400285  | 0.9210609940028851 | $10$ | -3.5083047578154947e-14 | 4.091858342398547e-13


Collaborations: Jia Cong Lin 
How do your errors compare to the error bounds for Taylor polynomials that we discussed in class?
According to the outputs as n increases the error calculated becomes smaller. For example when x = 4.0 and n = 2 the error is 0.0789390059971149. However, when n = 10 it becomes a smaller decimal 2.886091365894572e-11. When looking at the sin coding sample a difference can be seen from the first term to the last tem in which the absoulute error becomes smaller and smaller as the terms increase. 